### PR TITLE
Add QA cases for logs-agent cca_in_ad

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Adds a bunch of new QA cases for the logs-agent, generated based on the matrix of scenarios I've been working with for a while now.  It suggests running them all with cca_in_ad: true, and if that does something "weird" then comparing to a run with that value set to false.  

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tools/agent_QA/board.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import os
+
+from dotenv import load_dotenv
+from trello import TrelloClient
+
+
+class LocalBoard:
+    """A LocalBoard is like a Trello Board, but just prints cards to the console."""
```
Comment (gh123man) [inline tools/agent_QA/board.py:8]: 👌 I was thinking of adding something like this. This is great.

**Review by gh123man — APPROVED** [review]
LGTM. 

unrelated - but it looks like you are running into the linter too (which is very aggressive for non-product code like this IMO). I wonder if we should loosen the rules for this folder? 

Maybe it's just me but I had a hard time setting the linter rules up locally to behave like they do in CI, and not everything was able to be fixed with auto-formatting.

**Comment by djmitche** [comment]
Huh, yeah, it passed the pre-commit hooks :(

**Comment by djmitche** [comment]
I applied the patch from CI.  I think different versions of black apply different settings?

**Comment by djmitche** [comment]
Just Go intermittents that are red, so I'll merge..

**Pull request merged**